addons/test_mail/tests/test_mail_thread_internals.py

```diff
@@ -442,17 +442,12 @@ def test_suggested_recipients_default_create_value(self):
     @users("employee")
     def test_unlink_notification_message(self):
         channel = self.env['discuss.channel'].create({'name': 'testChannel'})
-        notification_msg = channel.with_user(self.user_admin).message_notify(
+        channel.with_user(self.user_admin).message_notify(
             body='test',
             partner_ids=[self.partner_2.id],
         )
-
-        with self.assertRaises(exceptions.AccessError):
-            notification_msg.with_env(self.env)._message_format(for_current_user=True)
```

Comment:
Just to clarify about removed check here:
- Method is private, access check is done before hand in real flow so this test is irrelevant.
- Test is a fake situation where message is created on the same transaction as the call to the method, which means message is already in cache so reading fields doesn't trigger the check again (in reality, it will).

In real flows:
- Either the message is read as sudo (guest, portal, etc) so the check always passes anyway.
- Or not sudo, and reading the fields of the message will triggered the check in a future commit (the `sudo` in formatter is not necessary and it will be removed in https://github.com/odoo/odoo/pull/171864 which requires more preparation on top of the current PR)

This method is anyway never called on messages for which access was not validated before-hand (in particular with built-in ACL of `search` to find the message (or pre-check with channel member-ship, portal token, ...), that alone is enough to justify removing the check.